File: backend/src/main/java/io/github/artemy/osipov/shop/controller/dto/DPresentItem.java
package io.github.artemy.osipov.shop.controller.dto;

import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.util.UUID;

@Data
public class DPresentItem {

    @NotNull
    @Valid
    private UUID candyId;

    @NotNull
    @Min(1)
    private Integer count;
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/controller/dto/DPresentItem.kt
package io.github.artemy.osipov.shop.controller.dto

import jakarta.validation.Valid
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotNull
import java.util.UUID

data class DPresentItem(
    @NotNull
    @Valid
    var candyId: UUID,

    @NotNull
    @Min(1)
    var count: Int
)